### ISSUE TITLE
Report disable errors the same way as lint errors

### ISSUE THREAD

**Issue by nex3** [stylelint/stylelint#4896]
> What is the problem you're trying to solve?

Currently, errors caused by `reportNeedlessDisables`, `reportInvalidScopeDisables`, and eventually `reportUnscopedDisables` (https://github.com/stylelint/stylelint/issues/2292) and `reportDisables` (https://github.com/stylelint/stylelint/issues/4875) are reported in a format that's inconsistent with (and less visually appealing than) the way normal lints are recorded.

> What solution would you like to see?

I propose that these disable issues instead be converted to the same object structure as lints, and be passed to the standard lint formatter to be reported appropriately. That way they'll automatically play nicely with both default and custom formatters, and look consistent with the most common reports.

**Comment by jeddy3** [comment]
Sounds good to me. I don't know why we didn't do that originally.

**Comment by edg2s** [comment]
Is there a reason these are styled with red x's in the output, instead of yellow warning triangles as is used for other warnings?

![image](https://user-images.githubusercontent.com/180672/99716109-9d2beb80-2a9f-11eb-93be-c44233b33639.png)

**Comment by nex3** [comment]
Stylelint makes the distinction between "warnings" (displayed with yellow triangles) and "errors" (displayed with red Xs). Normally, lints have a way of configuring which type they'll be reported as, but since these aren't configurable in the same way we had to make a choice—I chose errors, to match the idea that failing one of these requirements would cause the linter to produce a non-zero exit code.

**Comment by edg2s** [comment]
I get 0 exits codes with these in the command line:
```sh
$ ./node_modules/stylelint/bin/stylelint.js --rd src/styles/widgets/SelectFileWidget.less 

src/styles/widgets/SelectFileWidget.less
 21:4  ✖  Needless disable for "plugin/no-unsupported-browser-features"  --report-needless-disables

$ echo $?
0
```

Similarly, using the Node API, the "errored" flag is not set, and these are marked as warnings.

**Comment by nex3** [comment]
I'm surprised by that, but I'm afraid you've reached the limit of my stylelint expertise :woman_shrugging:. Maybe one of the more experienced maintainers could chime in?

**Comment by XhmikosR** [comment]
I opened #5046 this morning for the wrong error code.

**Comment by vankop** [comment]
As I see regression is here https://github.com/stylelint/stylelint/pull/4973

We need to hit this condition https://github.com/stylelint/stylelint/blob/0fdef35a3357563c21457b0d0463f1fcb058e512/lib/prepareReturnValue.js#L36
for 
https://github.com/stylelint/stylelint/blob/master/lib/cli.js#L563

fill free to send a PR

**Comment by jeddy3** [comment]
@vankop Thanks for identifying a possible fix!

Let's track the regression in @XhmikosR new issue: #5046
